README
NAME
    Net::SFTP::Foreign::Exceptional - DEPRECATED. Wraps Net::SFTP::Foreign
    to throw exceptions on failure.

VERSION
    version 0.011

DEPRECATION NOTICE
    This module is deprecated in favor of "Net::SFTP::Foreign's own error
    handling", specifically its "autodie" mode which offers the same
    functionality that this module was originally developed to achieve. You
    will receive a warning every time you call the "new()" method on this
    module.

SYNOPSIS
        use Net::SFTP::Foreign::Exceptional;

        my $sftp;
        eval { $sftp = Net::SFTP::Foreign::Exceptional->new(host => 'sftp.example.com'); 1 }
            or print "SFTP exception: $@\n";

DESCRIPTION
    Wrapper around Net::SFTP::Foreign that delegates all public method calls
    to it, throwing exceptions instead of merely returning "undef".

SUPPORT
  Perldoc
    You can find documentation for this module with the perldoc command.

      perldoc Net::SFTP::Foreign::Exceptional

  Websites
    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

    *   Search CPAN

        The default CPAN search engine, useful to view POD in HTML format.

        <http://search.cpan.org/dist/Net-SFTP-Foreign-Exceptional>

    *   AnnoCPAN

        The AnnoCPAN is a website that allows community annonations of Perl
        module documentation.

        <http://annocpan.org/dist/Net-SFTP-Foreign-Exceptional>

    *   CPAN Ratings

        The CPAN Ratings is a website that allows community ratings and
        reviews of Perl modules.

        <http://cpanratings.perl.org/d/Net-SFTP-Foreign-Exceptional>

    *   CPANTS

        The CPANTS is a website that analyzes the Kwalitee ( code metrics )
        of a distribution.

        <http://cpants.perl.org/dist/overview/Net-SFTP-Foreign-Exceptional>

    *   CPAN Testers

        The CPAN Testers is a network of smokers who run automated tests on
        uploaded CPAN distributions.

        <http://www.cpantesters.org/distro/N/Net-SFTP-Foreign-Exceptional>

    *   CPAN Testers Matrix

        The CPAN Testers Matrix is a website that provides a visual way to
        determine what Perls/platforms PASSed for a distribution.

        <http://matrix.cpantesters.org/?dist=Net-SFTP-Foreign-Exceptional>

    *   CPAN Testers Dependencies

        The CPAN Testers Dependencies is a website that shows a chart of the
        test results of all dependencies for a distribution.

        <http://deps.cpantesters.org/?module=Net::SFTP::Foreign::Exceptional
        >

  Bugs / Feature Requests
    Please report any bugs or feature requests through the web interface at
    <https://github.com/mjgardner/net-sftp-foreign-exceptional/issues>. You
    will be automatically notified of any progress on the request by the
    system.

  Source Code
    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    <https://github.com/mjgardner/net-sftp-foreign-exceptional>

      git clone git://github.com/mjgardner/net-sftp-foreign-exceptional.git

AUTHOR
    Mark Gardner <mjgardner@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2011 by GSI Commerce.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
{{$NEXT}}

0.011     2011-08-21 17:45:18 America/New_York
    deprecate in favor of Net::SFTP::Foreign constructor's autodie parameter
0.010     2011-08-16 14:22:18 America/New_York
    use Any::Moose for those who prefer Mouse-sized objects
0.005     2011-05-23 19:56:08 America/New_York
    use Moose itself rather than Class::Inspector, further reducing dependencies
    (DROLSKY++)
    add simple test for methods implemented
0.004     2011-05-20 12:35:08 America/New_York
    slightly fewer dependencies, only require perl 5.8 or later
0.003     2011-05-18 13:51:30 America/New_York
    wrap and delegate method calls rather than subclass (SALVA++)
    throw exceptions from the perspective of the caller (SALVA++)
0.002     2011-05-17 13:43:23 America/New_York
    require recent version of Net::SFTP::Foreign
0.001     2011-05-16 12:52:47 America/New_York
    initial release


lib/Net/SFTP/Foreign/Exceptional.pm
package Net::SFTP::Foreign::Exceptional;

use 5.008;
use strict;
use warnings;
use utf8;

# VERSION
use Carp;
use English '-no_match_vars';
use Any::Moose;
use Net::SFTP::Foreign 1.67;
use Package::DeprecationManager -deprecations =>
    { 'Net::SFTP::Foreign::Exceptional::new' => '0.11' };

our $DEPRECATION
    = 'Use the autodie parameter to Net::SFTP::Foreign directly.';

our @CARP_NOT
    = qw(Net::SFTP::Foreign Class::MOP::Method::Wrapped Mouse::Meta::Class);
my @METHODS = grep { $ARG ne 'new' and $ARG ne 'DESTROY' }
    map { $ARG->name }
    any_moose('::Meta::Class')->initialize('Net::SFTP::Foreign')
    ->get_all_methods();

has _sftp =>
    ( is => 'ro', isa => 'Net::SFTP::Foreign', handles => \@METHODS );

around BUILDARGS => sub {
    my ( $orig, $class ) = splice @ARG, 0, 2;

    deprecated($DEPRECATION);

    return $class->$orig(
        _sftp => Net::SFTP::Foreign->new( @ARG, autodie => 1 ) );
};

__PACKAGE__->meta->make_immutable();
1;

# ABSTRACT: DEPRECATED. Wraps Net::SFTP::Foreign to throw exceptions on failure.

=head1 SYNOPSIS

    use Net::SFTP::Foreign::Exceptional;

    my $sftp;
    eval { $sftp = Net::SFTP::Foreign::Exceptional->new(host => 'sftp.example.com'); 1 }
        or print "SFTP exception: $@\n";

=head1 DEPRECATION NOTICE

This module is deprecated in favor of
L<"Net::SFTP::Foreign's own error handling"|Net::SFTP::Foreign/"Error handling">,
specifically its C<autodie> mode which offers the same functionality that this
module was originally developed to achieve.  You will receive a warning every
time you call the C<new()> method on this module.

=head1 DESCRIPTION

Wrapper around L<Net::SFTP::Foreign|Net::SFTP::Foreign> that delegates all
public method calls to it, throwing exceptions instead of merely returning
C<undef>.


lib/Net/SFTP/Foreign/Exceptional/Error.pm
#
# This file is part of Net-SFTP-Foreign-Exceptional
#
# This software is copyright (c) 2011 by GSI Commerce.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
use utf8;
use Modern::Perl;    ## no critic (UselessNoCritic,RequireExplicitPackage)

package Net::SFTP::Foreign::Exceptional::Error;

BEGIN {
    $Net::SFTP::Foreign::Exceptional::Error::VERSION = '0.002';
}

use Moose;
use MooseX::Types::Moose 'Str';
use Net::SFTP::Foreign::Constants ':error';
use Net::SFTP::Foreign::Exceptional::Types 'SFTP_Error';
extends 'Throwable::Error';

has status_code => ( ro, required, isa => SFTP_Error );

has '+message' => ( ro, lazy,
    isa     => Str,
    default => sub { &{ $ARG[0]->status_code } },
);

1;

__END__

=pod

=for :stopwords Mark Gardner GSI Commerce cpan testmatrix url annocpan anno bugtracker rt
cpants kwalitee diff irc mailto metadata placeholders

=encoding utf8

=head1 NAME

Net::SFTP::Foreign::Exceptional::Error

=head1 VERSION

version 0.002

=head1 SUPPORT

=head2 Perldoc

You can find documentation for this module with the perldoc command.

  perldoc Net::SFTP::Foreign::Exceptional

=head2 Websites

The following websites have more information about this module, and may be of help to you. As always,
in addition to those websites please use your favorite search engine to discover more resources.

=over 4

=item *

Search CPAN

The default CPAN search engine, useful to view POD in HTML format.

L<http://search.cpan.org/dist/Net-SFTP-Foreign-Exceptional>

=item *

AnnoCPAN

The AnnoCPAN is a website that allows community annonations of Perl module documentation.

L<http://annocpan.org/dist/Net-SFTP-Foreign-Exceptional>

=item *

CPAN Ratings

The CPAN Ratings is a website that allows community ratings and reviews of Perl modules.

L<http://cpanratings.perl.org/d/Net-SFTP-Foreign-Exceptional>

=item *

CPANTS

The CPANTS is a website that analyzes the Kwalitee ( code metrics ) of a distribution.

L<http://cpants.perl.org/dist/overview/Net-SFTP-Foreign-Exceptional>

=item *

CPAN Testers

The CPAN Testers is a network of smokers who run automated tests on uploaded CPAN distributions.

L<http://www.cpantesters.org/distro/N/Net-SFTP-Foreign-Exceptional>

=item *

CPAN Testers Matrix

The CPAN Testers Matrix is a website that provides a visual way to determine what Perls/platforms PASSed for a distribution.

L<http://matrix.cpantesters.org/?dist=Net-SFTP-Foreign-Exceptional>

=item *

CPAN Testers Dependencies

The CPAN Testers Dependencies is a website that shows a chart of the test results of all dependencies for a distribution.

L<http://deps.cpantesters.org/?module=Net::SFTP::Foreign::Exceptional>

=back

=head2 Bugs / Feature Requests

Please report any bugs or feature requests through the web
interface at L<https://github.com/mjgardner/net-sftp-foreign-exceptional/issues>. You will be automatically notified of any
progress on the request by the system.

=head2 Source Code

The code is open to the world, and available for you to hack on. Please feel free to browse it and play
with it, or whatever. If you want to contribute patches, please send me a diff or prod me to pull
from your repository :)

L<https://github.com/mjgardner/net-sftp-foreign-exceptional>

  git clone git://github.com/mjgardner/net-sftp-foreign-exceptional.git

=head1 AUTHOR

Mark Gardner <mjgardner@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2011 by GSI Commerce.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/Net/SFTP/Foreign/Exceptional/Types.pm
#
# This file is part of Net-SFTP-Foreign-Exceptional
#
# This software is copyright (c) 2011 by GSI Commerce.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
use utf8;
use Modern::Perl;    ## no critic (UselessNoCritic,RequireExplicitPackage)

package Net::SFTP::Foreign::Exceptional::Types;

BEGIN {
    $Net::SFTP::Foreign::Exceptional::Types::VERSION = '0.002';
}

use Moose;
use MooseX::Types -declare => ['SFTP_Error'];
use MooseX::Types::Moose 'Value';
use List::MoreUtils 'any';
use Net::SFTP::Foreign::Constants;

subtype SFTP_Error, as Value, where {
    my $value = $ARG;
    any { $value == &{$ARG} }
    map {"Net::SFTP::Foreign::Constants::$ARG"}
        @{ $Net::SFTP::Foreign::Constants::EXPORT_TAGS{error} };
};

1;

__END__

=pod

=for :stopwords Mark Gardner GSI Commerce cpan testmatrix url annocpan anno bugtracker rt
cpants kwalitee diff irc mailto metadata placeholders

=encoding utf8

=head1 NAME

Net::SFTP::Foreign::Exceptional::Types

=head1 VERSION

version 0.002

=head1 SUPPORT

=head2 Perldoc

You can find documentation for this module with the perldoc command.

  perldoc Net::SFTP::Foreign::Exceptional

=head2 Websites

The following websites have more information about this module, and may be of help to you. As always,
in addition to those websites please use your favorite search engine to discover more resources.

=over 4

=item *

Search CPAN

The default CPAN search engine, useful to view POD in HTML format.

L<http://search.cpan.org/dist/Net-SFTP-Foreign-Exceptional>

=item *

AnnoCPAN

The AnnoCPAN is a website that allows community annonations of Perl module documentation.

L<http://annocpan.org/dist/Net-SFTP-Foreign-Exceptional>

=item *

CPAN Ratings

The CPAN Ratings is a website that allows community ratings and reviews of Perl modules.

L<http://cpanratings.perl.org/d/Net-SFTP-Foreign-Exceptional>

=item *

CPANTS

The CPANTS is a website that analyzes the Kwalitee ( code metrics ) of a distribution.

L<http://cpants.perl.org/dist/overview/Net-SFTP-Foreign-Exceptional>

=item *

CPAN Testers

The CPAN Testers is a network of smokers who run automated tests on uploaded CPAN distributions.

L<http://www.cpantesters.org/distro/N/Net-SFTP-Foreign-Exceptional>

=item *

CPAN Testers Matrix

The CPAN Testers Matrix is a website that provides a visual way to determine what Perls/platforms PASSed for a distribution.

L<http://matrix.cpantesters.org/?dist=Net-SFTP-Foreign-Exceptional>

=item *

CPAN Testers Dependencies

The CPAN Testers Dependencies is a website that shows a chart of the test results of all dependencies for a distribution.

L<http://deps.cpantesters.org/?module=Net::SFTP::Foreign::Exceptional>

=back

=head2 Bugs / Feature Requests

Please report any bugs or feature requests through the web
interface at L<https://github.com/mjgardner/net-sftp-foreign-exceptional/issues>. You will be automatically notified of any
progress on the request by the system.

=head2 Source Code

The code is open to the world, and available for you to hack on. Please feel free to browse it and play
with it, or whatever. If you want to contribute patches, please send me a diff or prod me to pull
from your repository :)

L<https://github.com/mjgardner/net-sftp-foreign-exceptional>

  git clone git://github.com/mjgardner/net-sftp-foreign-exceptional.git

=head1 AUTHOR

Mark Gardner <mjgardner@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2011 by GSI Commerce.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "DEPRECATED. Wraps Net::SFTP::Foreign to throw exceptions on failure.",
  "AUTHOR" => "Mark Gardner <mjgardner\@cpan.org>",
  "BUILD_REQUIRES" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Scalar::Util" => 0,
    "Test::Deep" => 0,
    "Test::More" => "0.94",
    "Test::Most" => 0
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Net-SFTP-Foreign-Exceptional",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Net::SFTP::Foreign::Exceptional",
  "PREREQ_PM" => {
    "Any::Moose" => 0,
    "Carp" => 0,
    "English" => 0,
    "Net::SFTP::Foreign" => "1.67",
    "Package::DeprecationManager" => 0,
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.011",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



